File: 2016/luke09.hs
import Data.List.Split
import qualified Data.Map as M

main :: IO ()
main = do transactions <- lines <$> readFile "luke09_data.txt"
          print . length . M.filter (>10) . balances $ splitOn "," <$> transactions

balances :: [[String]] -> M.Map String Integer
balances = foldl move M.empty
    where move acc [from, to, amountString] =
            let amount = read amountString :: Integer
                newAcc = M.insertWith (+) to amount acc
            in  if from == "None"
                then newAcc
                else M.insertWith (+) from (-amount) newAcc

File: 2016/luke09.rb
balances = File.readlines('luke09_data.txt').inject({}) { |res, line|
  from, to, amount = line.strip.split ","

  res[to] = (res[to] || 0) + amount.to_i
  res[from] -= amount.to_i unless from == "None"
  res
}

p balances.select { |_, v| v > 10 }.count
